test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -25,51 +28,162 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
 {
     public class OtlpLogExporterTests : Http2UnencryptedSupportTests
     {
-        [Fact]
-        public void ToOtlpLogRecordTest()
-        {
-            // Just a basic test to demonstrate an
-            // approach useful for testing.
-            // This needs to be expanded to
-            // actually test the conversion.
-            List<LogRecord> logRecords = new List<LogRecord>();
+        private readonly ILogger logger;
+        private readonly List<LogRecord> exportedItems = new();
+        private readonly ILoggerFactory loggerFactory;
+        private OpenTelemetryLoggerOptions options;
 
-            using var loggerFactory = LoggerFactory.Create(builder =>
+        public OtlpLogExporterTests()
+        {
+            this.loggerFactory = LoggerFactory.Create(builder =>
             {
                 builder.AddOpenTelemetry(options =>
                 {
-                    options.IncludeFormattedMessage = true;
-                    options.ParseStateValues = true;
-                    options.AddInMemoryExporter(logRecords);
+                    this.options = options;
+                    this.options.AddInMemoryExporter(this.exportedItems);
                 });
             });
 
-            // TODO:
-            // Validate attributes, severity, traceid,spanid etc.
+            this.logger = this.loggerFactory.CreateLogger<OtlpLogExporterTests>();
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordStateValues()
+        {
+            this.options.IncludeFormattedMessage = true;
+            this.options.ParseStateValues = true;
+
+            this.logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
+            Assert.Single(this.exportedItems);
 
-            var logger = loggerFactory.CreateLogger("log-category");
-            logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
-            Assert.Single(logRecords);
-            var logRecord = logRecords[0];
+            var logRecord = this.exportedItems[0];
             var otlpLogRecord = logRecord.ToOtlpLog();
             Assert.NotNull(otlpLogRecord);
             Assert.Equal("Hello from tomato 2.99.", otlpLogRecord.Body.StringValue);
```

Comment:
I think you should be able to also assert the state values have been added as attributes to the otlpLogRecord here.

Reply:
I've added the check for whether stateValues had been added as attributes. Thank you.